### PR TITLE
Repair build

### PR DESCRIPTION
**Pull request by doniks**

### CODE REVIEW

**Comment by doniks** [comment]
fixup https://github.com/ubports/docs.ubports.com/pull/627

**Comment by peat-psuwit** [comment]
Err... I'm pretty sure `apk` is not a mistake. `apk` is a package manager for Alpine Linux, and the package name seems to be correct. The real mistake is that `apt` support is removed for some reason.

Something like this should work:
```sh
  if command -v apt &> /dev/null; then
      install_via_apt
  elif command -v apk &> /dev/null; then
      install_via_apk
  elif command -v pacman &> /dev/null; then
      install_via_pacman
  else
    #...
  fi
```

**Comment by doniks** [comment]
Oh my god. Okok. I'll put back apk then. Thanks!

**Comment by doniks** [comment]
please recheck @peat-psuwit

**Comment by doniks** [comment]
the only variant I have tested my self is apt, but deleting `~/ubportsdocsenv` and then running `./build.sh` again and check that it creates the env ok and builds

File: build.sh [hunk 1]
```diff
@@ -31,26 +38,32 @@ install_via_pacman() {
...
   source $UBPORTSDOCSENV/bin/activate
 else
   echo -e "${YELLOW}Build environment not found in \"$UBPORTSDOCSENV\" ... creating it.${PLAIN}"
   echo -e "${YELLOW}Installing pip and virtualenv (using sudo).${PLAIN}"
-  # Check for apk or pacman
+  # Check for apt or pacman
   if command -v apk &> /dev/null; then
-      install_via_apk
+      echo "Detected apk"
+      sudo apk add python3 py3-pip py3-virtualenv
```
Comment (peat-psuwit) [inline build.sh:49]: Hmm if you're going to inline a code here, then remove the functions above? (Or vice-versa?)

**Comment by grenudi** [comment]
Sorry for the missing apt in the first place.
Here is polished version (i cant commit to PR without making a new one)
Tested on arch, i believe it will work in dev/ubuntu/alipine all the same. 
```sh
#!/bin/bash

set -Eeou pipefail

# color codes
GREEN='\033[0;32m'
YELLOW='\033[1;33m'
RED='\033[0;31m'
PLAIN='\033[0m'

# build directory
## use environment variable if it is set, or default value
UBPORTSDOCSENV=${UBPORTSDOCSENV-"$HOME/ubportsdocsenv"}
## use commandline parameter if it is given
UBPORTSDOCSENV=${1-$UBPORTSDOCSENV}

# DEPENDENCIES
APK_PACKAGES="python3 py3-pip py3-virtualenv"
APT_PACKAGES="python3-pip python3-virtualenv"
PACMAN_PACKAGES="python-pip python-virtualenv"

# FUNCTIONS
install_via_apk() {
    echo "Detected apk"
    echo "Installing packages: $APK_PACKAGES"
    sudo apk add $APK_PACKAGES
}
install_via_apt() {
    echo "Detected apt"
    echo "Installing packages: $APT_PACKAGES"
    sudo apt install $APT_PACKAGES
}
install_via_pacman() {
    echo "Detected pacman"
    echo "Installing packages: $PACMAN_PACKAGES"
    sudo pacman -S --noconfirm $PACMAN_PACKAGES
}
install_dependencies_for_most_linux_systems() {
    if command -v apk &> /dev/null; then
        install_via_apk
    elif command -v apt &> /dev/null; then
        install_via_apt
    elif command -v pacman &> /dev/null; then
        install_via_pacman
    else
        echo "Unsupported package manager. Please install the packages and create the virtualenv manually."
        exit 1
    fi
}



# check if virtualenv already exists
if [ -d "$UBPORTSDOCSENV" ]; then
  echo -e "${GREEN}Activating virtual build environment found in \"$UBPORTSDOCSENV\".${PLAIN}"
  source $UBPORTSDOCSENV/bin/activate
else
  echo -e "${YELLOW}Build environment not found in \"$UBPORTSDOCSENV\" ... creating it.${PLAIN}"
  echo -e "${YELLOW}Installing pip and virtualenv (using sudo).${PLAIN}"
  install_dependencies_for_most_linux_systems
  echo -e "${YELLOW}Creating and activating a virtual build environment in \"$UBPORTSDOCSENV\".${PLAIN}"
  virtualenv $UBPORTSDOCSENV
  source $UBPORTSDOCSENV/bin/activate
  echo -e "${YELLOW}Installing build tools and prerequisites (using pip).${PLAIN}"
  pip3 install -r requirements.txt
fi

echo -e "${GREEN}Building...${PLAIN}"
rm -rf _build/
if [ "$(uname)" == "Linux" ]; then
  sphinx-build -Wa . _build/html -j `nproc --all` || exit $?
else
  sphinx-build -Wa . _build/html -j `sysctl -n hw.ncpu` || exit $?
fi
sphinx-build -Wab rediraffecheckdiff . _build/html

echo "Open _build/html/index.html in your browser to navigate through the docs"
```

Test run from arch output:
```log
╭─dgn@uwu in repo: docs.ubports.com on  repair_build [!] via  v3.12.7 took 0s
╰─λ ./build.sh
Activating virtual build environment found in "/home/dgn/ubportsdocsenv".
Building...
Running Sphinx v8.0.2
loading translations [en]... done
making output directory... done
Converting `source_suffix = ['.rst']` to `source_suffix = {'.rst': 'restructuredtext'}`.
building [mo]: all of 0 po files
writing output...
building [html]: all source files
updating environment: [new config] 109 added, 0 changed, 0 removed
reading sources... [ 62%] porting/configure_test_fix/Bluetooth .. porting/configure_test_fix/device_info/HfdSerreading sources... [100%] userguide/install
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
copying assets...
copying downloadable files... [100%] userguide/advanceduse/files/mircast.sh
copying static files... done
copying extra files... done
copying assets: done
writing output... [100%] userguide/advanceduse/screencasting .. userguide/install
generating indices... genindex done
writing additional pages... search done
copying images... [100%] _static/images/userguide/volte-indicator.png
dumping search index in English (code: en)... done
dumping object inventory... done
Writing redirects...
(good) porting/index.html --> porting/introduction/index.html
(good) porting/introduction.html --> porting/introduction/index.html
(good) porting/halium_7-1/index.html --> porting/introduction/index.html
(good) porting/halium_7-1/Building.html --> porting/build_and_boot/index.html
(good) porting/halium_7-1/Common_errors.html --> porting/build_and_boot/index.html
(good) porting/halium_7-1/Installing.html --> porting/build_and_boot/index.html
(good) porting/halium_7-1/Unity8.html --> porting/configure_test_fix/index.html
(good) porting/halium_7-1/Configuring.html --> porting/configure_test_fix/index.html
(good) porting/halium_9/Halium_9_GSI_porting.html --> porting/build_and_boot/index.html
(good) porting/halium_9/Halium_9_common_errors.html --> porting/build_and_boot/index.html
(good) porting/halium_9/index.html --> porting/build_and_boot/index.html
(good) porting/halium_7-1/Reference_material.html --> porting/configure_test_fix/index.html
(good) porting/build_and_boot/H9_gitlab_build.html --> porting/build_and_boot/standalone_kernel_build.html
(good) porting/UBports_installer.html --> porting/finalize/UBports_installer.html
(good) porting/UBports_recovery.html --> porting/finalize/UBports_recovery.html
(good) porting/Preparations.html --> porting/introduction/Preparations.html
(good) appdev/getting-started.html --> appdev/index.html
(good) appdev/publishing/index.html --> appdev/index.html
(good) appdev/samples/index.html --> appdev/index.html
(good) appdev/guides/cookbook.html --> appdev/guides/index.html
(good) userguide/dailyuse/anbox.html --> userguide/dailyuse/waydroid.html
(good) about/process/ppa.html --> about/process/branch-naming.html
build succeeded.

The HTML pages are in _build/html.
Running Sphinx v8.0.2
loading translations [en]... done
Converting `source_suffix = ['.rst']` to `source_suffix = {'.rst': 'restructuredtext'}`.
loading pickled environment... done
building [mo]: all of 0 po files
writing output...
building [rediraffecheckdiff]: all source files
looking for now-outdated files... none found
preparing documents... done
copying assets...
copying assets: done
writing output... [100%] userguide/install
rediraffe: Redirect generation skipped for unsupported builders. Supported builders: html, dirhtml, readthedocs, readthedocsdirhtml.
build succeeded.
Open _build/html/index.html in your browser to navigate through the docs

╭─dgn@uwu in repo: docs.ubports.com on  repair_build [!] via  v3.12.7 took 3s
```

**Pull request merged**